cypress/integration/components/dataTable.js

```diff
@@ -0,0 +1,204 @@
+  });
+
+  context('external table controls', () => {
+    it('should update pagination style to basic', () => {
+      cy.get(`${exampleContainerSelector} novo-data-table novo-data-table-pagination`)
+        .eq(0)
+        .should('have.class', 'standard');
+      cy.get(`${exampleContainerSelector} novo-tiles [data-automation-id="Basic"]`)
+        .eq(0)
+        .should('not.have.class', 'active')
+        .click()
+        .should('have.class', 'active');
+      cy.get(`${exampleContainerSelector} novo-data-table novo-data-table-pagination`)
+        .should('have.class', 'basic');
+    });
+    it('should show global search', () => {
+      cy.get(`${exampleContainerSelector} novo-data-table novo-search`).should('not.exist');
+      cy.get(`${exampleContainerSelector} novo-tiles [data-automation-id="Show"]`)
+        .eq(0)
+        .should('not.have.class', 'active')
+        .click()
+        .should('have.class', 'active');
+      cy.get(`${exampleContainerSelector} novo-data-table novo-search`).should('exist');
+    });
+    it('should be able to configure columns', () => {
+      //TODO
+
+    });
+    it('should show and hide all row details', () => {
+      cy.get(`${exampleContainerSelector} novo-data-table novo-data-table-expand-header-cell > i`)
+        .should('not.have.class', 'expanded');
+      cy.get(`${exampleContainerSelector} button`)
+        .eq(1).click();
+      cy.get(`${exampleContainerSelector} novo-data-table novo-data-table-expand-header-cell > i`)
+        .should('have.class', 'expanded');
+      cy.get(`${exampleContainerSelector} button`)
+        .eq(2).click();
+      cy.get(`${exampleContainerSelector} novo-data-table novo-data-table-expand-header-cell > i`)
+        .should('not.have.class', 'expanded');
+    });
+  });
+  context('pagination', () => {
+    it('should update data table row count', () => {
+      const paginationOptions = [10,50,100,250,500];
+      cy.get(`${exampleContainerSelector} novo-tiles [data-automation-id="Basic"]`)
+        .eq(0)
+        .click()
+        .should('have.class', 'active');
+      cy.get(`${exampleContainerSelector} novo-data-table novo-data-table-pagination`)
+        .should('have.class', 'basic');
+      paginationOptions.forEach((option)=> {
+        cy.get(`${exampleContainerSelector} novo-data-table novo-data-table-pagination [data-automation-id="${option}"]`)
+          .eq(0)
+          .click()
+          .should('have.class', 'active');
+        cy.get(`${exampleContainerSelector} novo-data-table novo-data-table-row`)
+          .should('have.length', option);
+      })
+    });
+    it('should switch to the next page properly', () => {
+      cy.get(`${exampleContainerSelector} novo-tiles [data-automation-id="Basic"]`)
+        .eq(0)
+        .click()
+        .should('have.class', 'active');
+      cy.get(`${exampleContainerSelector} novo-data-table novo-data-table-pagination`)
+        .should('have.class', 'basic');
```

Comment:
this is great! do we want to organize the code similar to how novo automation is done? so that we separate our the selectors into page classes, etc.

Reply:
I _would_ agree, however due to how our demo is written, there is not much reuse of code outside of the file for the particular component/element.

Reply:
it just makes it easier to read the code without worrying about which selector to change where as opposed to being able to reuse code.